cluster-autoscaler/core/scaleup/orchestrator/async_initializer.go

```diff
@@ -84,12 +94,17 @@ func (s *asyncNodeGroupInitializer) InitializeNodeGroup(result nodegroups.AsyncN
 	nodeInfos := make(map[string]*framework.NodeInfo)
 	var scaleUpInfos []nodegroupset.ScaleUpInfo
 	for _, nodeGroup := range result.CreationResult.AllCreatedNodeGroups() {
-		if targetSize := result.TargetSizes[nodeGroup.Id()]; targetSize > 0 {
+		upcomingId, ok := result.CreatedToUpcomingMapping[nodeGroup.Id()]
+		if !ok {
+			klog.Errorf("Couldn't retrieve initialization data for new node group %v. It won't get initialized.", nodeGroup.Id())
+			continue
+		}
+		if targetSize := s.allTargetSizes[upcomingId]; targetSize > 0 {
```

Comment:
shouldn't we invoke the new `s.GetTargetSize(upcomingId)` here?

Reply:
Thanks, done.